### DISCUSSION TITLE
Docker choose Python version

### DISCUSSION THREAD

**Discussion by thomaspoignant** [renovatebot/renovate#10676] [Request Help]
Hello 👋 

I try to use the docker image for renovate and I have an error while running it because the version in my `Pipenv` file is not the same as the one in the container.

```
"Command failed: pipenv lock\nWarning: Python 3.7 was not found on your system...\nNeither 'pyenv' nor 'asdf' could be found to install Python.\nYou can specify specific versions of Python with:\n$ pipenv --python path/to/python\n"
```

Any idea on how to deal with that?

**Comment by rarkins** [comment]
How are you running renovate?

**Reply by thomaspoignant** [reply]
Like this :
```shell
docker run -e RENOVATE_TOKEN="$GITHUB_TOKEN" -v $PWD/test.log:/usr/src/app/test.log -v $PWD/config.js:/usr/src/app/config.js renovate/renovate <my repo>

```

**Reply by rarkins** [reply]
The Docker image comes with latest python so you'd need to build a custom image if you want to use python 3.7

**Reply by thomaspoignant** [reply]
I can't find any docs on how to build it.
Should I use https://github.com/renovatebot/docker-buildpack ?

**Reply by thomaspoignant** [reply]
or https://github.com/renovatebot/docker-python ?

**Reply by rarkins** [reply]
You can use https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile and change or any lines you need there

**Comment by viceice** [comment]
Better use slim renovate image and mount the docker socket, so renovate can create sidecar container to run python in required version.

see https://github.com/renovatebot/docker-renovate#samples

**Reply by Nebucatnetzer** [reply]
How can I specify the required Python version?
E.g. I'm running Ansible on Ubuntu 18.04 which comes with Python 3.6
We have the same "problem" PHP.
I search the documentation for an option but there doesn't seem to be something like `interpreterVersion`.
We are running the slim image.

**Reply by Nebucatnetzer** [reply]
Found it for Python:
https://docs.renovatebot.com/configuration-options/#constraints
